src/diem/offchain/types/schema.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+
+from marshmallow import Schema, fields, validate, post_load, post_dump
+from . import data_types
+
+import typing
+
+
+T = typing.TypeVar("T")
+
+
+class BaseSchema(Schema, typing.Generic[T]):
+    @post_dump
+    def remove_none_values(
+        self, data: typing.Dict[str, typing.Any], **kwargs  # pyre-ignore
+    ) -> typing.Dict[str, typing.Any]:
+        return {key: value for key, value in data.items() if value is not None}
+
+    @post_load
+    def make_obj(self, data: typing.Dict[str, typing.Any], **kwargs) -> T:  # pyre-ignore
+        base = type(self).__orig_bases__[0]  # pyre-ignore
+        cls = typing.get_args(base)[0]
+        return cls(**data)
+
+
+class NationalIdObject(BaseSchema[data_types.NationalIdObject]):
+    id_value = fields.Str(allow_none=True)
```

Comment:
id_value can't be none
https://github.com/diem/lip/blob/master/lips/lip-1.mdx#nationalidobject